pytorch_pfn_extras/onnx/unstrip_tensor.py

```diff
@@ -0,0 +1,131 @@
+import argparse
+import json
+import math
+from pathlib import Path
+
+import onnx
+import onnx.numpy_helper
+import numpy
+
+from pytorch_pfn_extras.onnx import strip_large_tensor as strip
+
+
+def _unstrip_tensor(tensor: onnx.TensorProto) -> None:
+    meta_dict = {}
+    meta_dict_idx = 0
+    for i, external_data in enumerate(tensor.external_data):
+        if external_data.key != "location":
+            continue
+        try:
+            external_data_dict = json.loads(external_data.value)
+            if external_data_dict.get("type", "") == "stripped":
+                meta_dict = external_data_dict
+                meta_dict_idx = i
+                break
+        except ValueError:
+            continue
+    if not meta_dict:
+        return None
+    ave = meta_dict.get("average", None)
+    var = meta_dict.get("variance", None)
+    if ave is None or var is None:
+        return None
+
+    np_dtype = onnx.mapping.TENSOR_TYPE_TO_NP_TYPE[tensor.data_type]
+    dummy_array = numpy.random.normal(ave, math.sqrt(var), tensor.dims).astype(np_dtype)
+    dummy_tensor = onnx.numpy_helper.from_array(dummy_array)
+    tensor.data_location = onnx.TensorProto.DEFAULT
+    tensor.raw_data = dummy_tensor.raw_data
+    del tensor.external_data[meta_dict_idx]
+
+
+def _unstrip_graph(graph: onnx.GraphProto) -> None:
+    for init in graph.initializer:
+        if not strip._is_stripped_or_set_external(init):
+            continue
+        _unstrip_tensor(init)
+
+    for node in graph.node:
+        for attr in node.attribute:
+            if attr.HasField("g"):
+                _unstrip_graph(attr.g)
+
+
+def _unstrip_onnx_from_path(path: Path) -> onnx.GraphProto:
+    onnx_graph = onnx.load(path, load_external_data=False)
+    _unstrip_graph(onnx_graph.graph)
+    return onnx_graph
+
+
+def _unstrip_tensor_from_path(path: Path) -> onnx.TensorProto:
+    onnx_tensor = onnx.TensorProto()
+    with open(path, "rb") as fp:
+        onnx_tensor.ParseFromString(fp.read())
+    _unstrip_tensor(onnx_tensor)
+    return onnx_tensor
+
+
+def unstrip(path: str, out_path: str = "") -> None:
+    """Unstrip ONNX models and test data(.pb).
+
+    Add tensor(raw data) to the target ONNXs (and test data).
+    Values are random following mean and variance written in meta information.
+
+    Args:
+        path (str): The target directory path, ONNX file, or Tensor (Protobuf)
+            file path.
+        out_path (str): Output path to be written.
+    """
+    target_path = Path(path)
+    if not target_path.exists():
+        print(f"Error: the target path is not found, {path}")
+        return None
+    written_path = Path(out_path if out_path else path)
+
+    if target_path.is_dir():
```

Comment:
https://github.com/disktnk/pytorch-pfn-extras/pull/1 meta.json should be also updated

Reply:
supported to handle `meta.json` if existed